hypershift-operator/controllers/nodepool/conditions.go

```diff
@@ -824,3 +827,97 @@ func (r NodePoolReconciler) createValidGeneratedPayloadCondition(ctx context.Con
 
 	return condition, nil
 }
+
+// createValidGeneratedPayloadCondition creates a condition for the NodePool based on the tokenSecret data.
+func (r NodePoolReconciler) validPlatformConfigCondition(ctx context.Context, nodePool *hyperv1.NodePool, hc *hyperv1.HostedCluster) (*ctrl.Result, error) {
+	condition := &hyperv1.NodePoolCondition{
+		Type:               hyperv1.NodePoolValidPlatformConfigConditionType,
+		Status:             corev1.ConditionTrue,
+		Reason:             hyperv1.AsExpectedReason,
+		Message:            hyperv1.AllIsWellMessage,
+		ObservedGeneration: nodePool.Generation,
+	}
+
+	// TODO: add validation for other platforms
+	switch nodePool.Spec.Platform.Type {
+	case hyperv1.AWSPlatform:
+		err := r.validateAWSPlatformConfig(ctx, nodePool, hc)
+		if err != nil {
+			condition.Status = corev1.ConditionFalse
+			condition.Reason = hyperv1.AWSErrorReason
+			condition.Message = err.Error()
+		}
+	}
+
+	SetStatusCondition(&nodePool.Status.Conditions, *condition)
+	return nil, nil
+}
+
+func (r NodePoolReconciler) validateAWSPlatformConfig(ctx context.Context, nodePool *hyperv1.NodePool, hc *hyperv1.HostedCluster) error {
```

Comment:
let's move any platform specific implementation to its own file i.e. aws.go

Reply:
done